server/src/main/java/com/google/musicanalysis/site/OAuthCallbackServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.musicanalysis.site;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.JsonParser;
+
+@WebServlet("/api/oauth/callback")
+public class OAuthCallbackServlet extends HttpServlet {
```

Comment:
**This servlet requests refresh and access tokens; Spotify returns access and refresh tokens**

[see step 2 for more info](https://developer.spotify.com/documentation/general/guides/authorization-guide/)